[Mappings editor] Add support for constant_keyword field type

alisonelizabeth opened the pull request:
Fixes https://github.com/elastic/kibana/issues/69751

### Release note
The mappings editor in the Index Templates UI now supports configuring the [constant_keyword field type](https://www.elastic.co/guide/en/elasticsearch/reference/master/keyword.html#constant-keyword-field-type).

<img width="740" alt="Screen Shot 2020-09-02 at 4 01 29 PM" src="https://user-images.githubusercontent.com/5226211/92031569-cc4f7e00-ed36-11ea-8a00-d06dbe9cf2ae.png">

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

debadair commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/constant_keyword_type.tsx:
```diff
@@ -0,0 +1,81 @@
+            'xpack.idxMgmt.mappingsEditor.constantKeyword.valueFieldDescription',
+            {
+              defaultMessage:
+                'The value to associate with all documents in the index. If this parameter is not provided, it is set based on the first document that gets indexed.',
```
```suggestion
                'The value of this field for all documents in the index. If not specified, defaults to the value specified in the first document indexed.',
```

debadair commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/constant_keyword_type.tsx:
```diff
@@ -0,0 +1,81 @@
+        {/* Meta field */}
+        <EditFieldFormRow
+          title={i18n.translate('xpack.idxMgmt.mappingsEditor.constantKeyword.metaFieldTitle', {
+            defaultMessage: 'Set meta',
```
I know this is the name of the parameter, but I'd be inclined to label is as "Set metadata"

debadair commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/constant_keyword_type.tsx:
```diff
@@ -0,0 +1,81 @@
+          description={i18n.translate(
+            'xpack.idxMgmt.mappingsEditor.constantKeyword.metaFieldDescription',
+            {
+              defaultMessage: 'Metadata about the field.',
```
```suggestion
              defaultMessage: 'Arbitrary information about the field. Specify as JSON key-value pairs.',
```
```suggestion
              defaultMessage: 'Metadata about the field.',
```

debadair commented on x-pack/plugins/index_management/public/application/components/mappings_editor/constants/parameters_definition.tsx:
```diff
@@ -404,6 +404,63 @@ export const PARAMETERS_DEFINITION: { [key in ParameterName]: ParameterDefinitio
+          id="xpack.idxMgmt.mappingsEditor.parameters.metaHelpText"
+          defaultMessage="Use JSON format: {code}"
+          values={{
+            code: <EuiCode>{JSON.stringify({ arbitrary_data: 'anything_goes' })}</EuiCode>,
```
```suggestion
            code: <EuiCode>{JSON.stringify({ arbitrary_key: 'anything_goes' })}</EuiCode>,
```

debadair commented on x-pack/plugins/index_management/public/application/components/mappings_editor/constants/parameters_definition.tsx:
```diff
@@ -404,6 +404,63 @@ export const PARAMETERS_DEFINITION: { [key in ParameterName]: ParameterDefinitio
+        {
+          validator: isJsonField(
+            i18n.translate('xpack.idxMgmt.mappingsEditor.parameters.metaFieldEditorJsonError', {
+              defaultMessage: 'The meta JSON is not valid.',
```
I think I'd just say "Invalid JSON."

debadair commented on x-pack/plugins/index_management/public/application/components/mappings_editor/constants/data_types_definition.tsx:
```diff
@@ -71,6 +71,26 @@ export const TYPE_DEFINITION: { [key in DataType]: DataTypeDefinition } = {
+      <p>
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.dataType.constantKeywordLongDescription"
+          defaultMessage="Constant keyword fields support the same queries and aggregations as {keyword} fields, but take advantage of the fact that all documents have the same value per index."
```
I'd lead with a definition of the field type. 

```suggestion
          defaultMessage="A special type of keyword field for fields that contain the same keyword across all documents in the index. Supports the same queries and aggregations as {keyword} fields."
```

alisonelizabeth replied:
We've typically started all the descriptions with the name of the field type, so I've changed this to: 

> Constant keyword fields are a special type of keyword field for fields that contain the same keyword across all documents in the index. Supports the same queries and aggregations as {keyword} fields."

debadair reviewed (approved):
Left a few suggestions. I didn't look closely at the other field types, so if they are inconsistent with existing UI text, I'd err on the side of consistency. (Though it looks like an editing pass of all of them might make sense at some point.)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/)
* Commit: df653c575494ed3b33b32efcdbf17b95887e8f87

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/df653c575494ed3b33b32efcdbf17b95887e8f87) | diff | [baseline](https://github.com/elastic/kibana/commit/71b9dedfc4eadd97db2b6b0dc038d0f640babf7b) |
| --- | --- | --- | --- |
| indexManagement | 501 | +1 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/df653c575494ed3b33b32efcdbf17b95887e8f87) | diff | [baseline](https://github.com/elastic/kibana/commit/71b9dedfc4eadd97db2b6b0dc038d0f640babf7b) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +4.8KB | 1.6MB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/","number":"72037","commit":"df653c575494ed3b33b32efcdbf17b95887e8f87"}]}
PIPELINE-->

alisonelizabeth commented:
Thanks @debadair for the copy review!

> Left a few suggestions. I didn't look closely at the other field types, so if they are inconsistent with existing UI text, I'd err on the side of consistency. (Though it looks like an editing pass of all of them might make sense at some point.)

👍 The field type forms did go through copy review when they were initially added, but doesn't hurt to do a second pass at some point.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/)
* Commit: ac1179803342c07b8a2a0d24394b5e0bb206984e

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/ac1179803342c07b8a2a0d24394b5e0bb206984e) | diff | [baseline](https://github.com/elastic/kibana/commit/71b9dedfc4eadd97db2b6b0dc038d0f640babf7b) |
| --- | --- | --- | --- |
| indexManagement | 501 | +1 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/ac1179803342c07b8a2a0d24394b5e0bb206984e) | diff | [baseline](https://github.com/elastic/kibana/commit/71b9dedfc4eadd97db2b6b0dc038d0f640babf7b) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +4.8KB | 1.6MB |

### History
* :green_heart: [Build #72037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/) succeeded df653c575494ed3b33b32efcdbf17b95887e8f87

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/","number":"72037","commit":"df653c575494ed3b33b32efcdbf17b95887e8f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/","number":"72358","commit":"ac1179803342c07b8a2a0d24394b5e0bb206984e"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/)
* Commit: 7cc25b8b3e3335271a28aa7e0dadf4380306053f

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/7cc25b8b3e3335271a28aa7e0dadf4380306053f) | diff | [baseline](https://github.com/elastic/kibana/commit/db67eb0b29bbc794dccfeb64615c317818cf39c8) |
| --- | --- | --- | --- |
| indexManagement | 501 | +1 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/7cc25b8b3e3335271a28aa7e0dadf4380306053f) | diff | [baseline](https://github.com/elastic/kibana/commit/db67eb0b29bbc794dccfeb64615c317818cf39c8) |
| --- | --- | --- | --- |
| indexManagement | 1.5MB | +4.8KB | 1.5MB |

### History
* :green_heart: [Build #72358](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/) succeeded ac1179803342c07b8a2a0d24394b5e0bb206984e
* :green_heart: [Build #72037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/) succeeded df653c575494ed3b33b32efcdbf17b95887e8f87

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/","number":"72037","commit":"df653c575494ed3b33b32efcdbf17b95887e8f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/","number":"72358","commit":"ac1179803342c07b8a2a0d24394b5e0bb206984e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/","number":"72739","commit":"7cc25b8b3e3335271a28aa7e0dadf4380306053f"}]}
PIPELINE-->

sebelga commented:
@elasticmachine merge upstream

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/constants/parameters_definition.tsx:
```diff
@@ -404,6 +404,63 @@ export const PARAMETERS_DEFINITION: { [key in ParameterName]: ParameterDefinitio
+        return JSON.stringify(value, null, 2);
+      },
+      serializer: (value: string) => {
+        try {
```
Same comment here as in https://github.com/elastic/kibana/pull/76671/. No need anymore of the `try/catch`

```js
serializer: (value: string) => {
  const parsed = JSON.parse(value);
  // If an empty object was passed, strip out this value entirely.
  if (!Object.keys(parsed).length) {
    return undefined;
  }
  return parsed;
},
```

sebelga reviewed (approved):
Great work @alisonelizabeth ! I tested locally and it works as expected. The issue I mentioned in https://github.com/elastic/kibana/pull/76671 around meta does not happen here.

But I would add an additional `allowArray` in the `isJsonField` validator (`true` by default to not create breaking change) that would return an error if an array is provided and the option is set to `false`.

sebelga commented:
[EDIT]

The isue with the `meta` also occurs here as per my comment https://github.com/elastic/kibana/pull/76671#issuecomment-688219343

It seems to occur when we use an index pattern that could also match other template, even with a totally unique field defined on the new template. And removing the meta fixes the ES error. I think that this should be investigated further.

sebelga commented:
[EDIT 2]

Actually the index pattern does not have an effect. This request fails as well

<img width="873" alt="Screenshot 2020-09-07 at 12 09 45" src="https://user-images.githubusercontent.com/2854616/92376631-29ff1400-f103-11ea-9673-a2b2fea3f8f6.png">

sebelga reviewed (commented):
I'd like to investigate a bit further before approving as there is a very odd behavior. Maybe it is 100% on ES side, but I'd like to confirm this.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/)
* Commit: d6c3dc6c485ad1bdcce8e6616360eaea9b65a442

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/d6c3dc6c485ad1bdcce8e6616360eaea9b65a442) | diff | [baseline](https://github.com/elastic/kibana/commit/b65c95a7d89e1e09b5082e351aace7e292dde7c6) |
| --- | --- | --- | --- |
| indexManagement | 501 | +1 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/d6c3dc6c485ad1bdcce8e6616360eaea9b65a442) | diff | [baseline](https://github.com/elastic/kibana/commit/b65c95a7d89e1e09b5082e351aace7e292dde7c6) |
| --- | --- | --- | --- |
| indexManagement | 1.5MB | +4.8KB | 1.5MB |

### History
* :green_heart: [Build #72739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/) succeeded 7cc25b8b3e3335271a28aa7e0dadf4380306053f
* :green_heart: [Build #72358](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/) succeeded ac1179803342c07b8a2a0d24394b5e0bb206984e
* :green_heart: [Build #72037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/) succeeded df653c575494ed3b33b32efcdbf17b95887e8f87

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/","number":"72037","commit":"df653c575494ed3b33b32efcdbf17b95887e8f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/","number":"72358","commit":"ac1179803342c07b8a2a0d24394b5e0bb206984e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/","number":"72739","commit":"7cc25b8b3e3335271a28aa7e0dadf4380306053f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/","number":"72862","commit":"d6c3dc6c485ad1bdcce8e6616360eaea9b65a442"}]}
PIPELINE-->

alisonelizabeth commented:
@sebelga thanks for the review! I believe I addressed your feedback. Can you take another look when you get a chance?

The `meta` field only accepts strings as values. I believe that is why you were hitting the error. 

> Field metadata enforces at most 5 entries, that keys have a length that is less than or equal to 20, and that values are strings whose length is less than or equal to 50.

Do you think we should enforce these rules on the client? It is unfortunate that ES doesn't return a better error message.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74199/)
* Commit: f919daae4bce1891e30ee5c878862fbadc236bbc

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/f919daae4bce1891e30ee5c878862fbadc236bbc) | diff | [baseline](https://github.com/elastic/kibana/commit/7dca537382a830b8f383d7429be6e25edf69ab7d) |
| --- | --- | --- | --- |
| indexManagement | 504 | +1 | 503 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/f919daae4bce1891e30ee5c878862fbadc236bbc) | diff | [baseline](https://github.com/elastic/kibana/commit/7dca537382a830b8f383d7429be6e25edf69ab7d) |
| --- | --- | --- | --- |
| indexManagement | 1.5MB | +5.1KB | 1.5MB |

### History
* :green_heart: [Build #72862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/) succeeded d6c3dc6c485ad1bdcce8e6616360eaea9b65a442
* :green_heart: [Build #72739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/) succeeded 7cc25b8b3e3335271a28aa7e0dadf4380306053f
* :green_heart: [Build #72358](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/) succeeded ac1179803342c07b8a2a0d24394b5e0bb206984e
* :green_heart: [Build #72037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/) succeeded df653c575494ed3b33b32efcdbf17b95887e8f87

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/","number":"72037","commit":"df653c575494ed3b33b32efcdbf17b95887e8f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/","number":"72358","commit":"ac1179803342c07b8a2a0d24394b5e0bb206984e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/","number":"72739","commit":"7cc25b8b3e3335271a28aa7e0dadf4380306053f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/","number":"72862","commit":"d6c3dc6c485ad1bdcce8e6616360eaea9b65a442"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74199/","number":"74199","commit":"f919daae4bce1891e30ee5c878862fbadc236bbc"}]}
PIPELINE-->

sebelga commented:
I will review it later today or tm morning.

> The meta field only accepts strings as values. I believe that is why you were hitting the error.

I do think it'd be great to have a validator that makes sure that each value is a string. It is easy to add on our side and can be a long debugging for the user to understand why.

Maybe the reason is better explained by ES deeply nested in the error. I opened https://github.com/elastic/kibana/issues/77165 to improve the error message surfaced in the UI.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74788/)
* Commit: 22caf7937634837e37e9f5a05fa7e585ef395740

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/22caf7937634837e37e9f5a05fa7e585ef395740) | diff | [baseline](https://github.com/elastic/kibana/commit/7dca537382a830b8f383d7429be6e25edf69ab7d) |
| --- | --- | --- | --- |
| indexManagement | 504 | +1 | 503 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/22caf7937634837e37e9f5a05fa7e585ef395740) | diff | [baseline](https://github.com/elastic/kibana/commit/7dca537382a830b8f383d7429be6e25edf69ab7d) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +5.5KB | 1.5MB |

### History
* :green_heart: [Build #74199](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74199/) succeeded f919daae4bce1891e30ee5c878862fbadc236bbc
* :green_heart: [Build #72862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/) succeeded d6c3dc6c485ad1bdcce8e6616360eaea9b65a442
* :green_heart: [Build #72739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/) succeeded 7cc25b8b3e3335271a28aa7e0dadf4380306053f
* :green_heart: [Build #72358](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/) succeeded ac1179803342c07b8a2a0d24394b5e0bb206984e
* :green_heart: [Build #72037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/) succeeded df653c575494ed3b33b32efcdbf17b95887e8f87

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72037/","number":"72037","commit":"df653c575494ed3b33b32efcdbf17b95887e8f87"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/","number":"72358","commit":"ac1179803342c07b8a2a0d24394b5e0bb206984e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/","number":"72739","commit":"7cc25b8b3e3335271a28aa7e0dadf4380306053f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/","number":"72862","commit":"d6c3dc6c485ad1bdcce8e6616360eaea9b65a442"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74199/","number":"74199","commit":"f919daae4bce1891e30ee5c878862fbadc236bbc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74788/","number":"74788","commit":"22caf7937634837e37e9f5a05fa7e585ef395740"}]}
PIPELINE-->

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74918/)
* Commit: 5c75148fe027c14678e70436ce2857fe12ec9297

### Build metrics
> ‼️ unable to find a baseline build for [master@[e5b302e](https://github.com/elastic/kibana/commit/e5b302eeee4d7a9331c56c5ae93f7e8603603754)]. Try merging the upstream branch and trying again.

### History
* :green_heart: [Build #74788](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74788/) succeeded 22caf7937634837e37e9f5a05fa7e585ef395740
* :green_heart: [Build #74199](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74199/) succeeded f919daae4bce1891e30ee5c878862fbadc236bbc
* :green_heart: [Build #72862](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/) succeeded d6c3dc6c485ad1bdcce8e6616360eaea9b65a442
* :green_heart: [Build #72739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/) succeeded 7cc25b8b3e3335271a28aa7e0dadf4380306053f
* :green_heart: [Build #72358](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/) succeeded ac1179803342c07b8a2a0d24394b5e0bb206984e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72358/","number":"72358","commit":"ac1179803342c07b8a2a0d24394b5e0bb206984e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72739/","number":"72739","commit":"7cc25b8b3e3335271a28aa7e0dadf4380306053f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72862/","number":"72862","commit":"d6c3dc6c485ad1bdcce8e6616360eaea9b65a442"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74199/","number":"74199","commit":"f919daae4bce1891e30ee5c878862fbadc236bbc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74788/","number":"74788","commit":"22caf7937634837e37e9f5a05fa7e585ef395740"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74918/","number":"74918","commit":"5c75148fe027c14678e70436ce2857fe12ec9297"}]}
PIPELINE-->

sebelga reviewed (approved):
Thanks for making the changes and adding the validation @alisonelizabeth ! Tested locally and works as expected 👍
